pkg/issuer/acme/http/http.go

```diff
@@ -125,6 +125,9 @@ func (s *Solver) Present(ctx context.Context, issuer v1.GenericIssuer, ch *cmacm
 			return utilerrors.NewAggregate([]error{podErr, svcErr, ingressErr})
 		}
 		if ch.Spec.Solver.HTTP01.GatewayHTTPRoute != nil {
+			if !s.GatewaySolverEnabled {
+				return fmt.Errorf("could't Present challenge %s/%s: gateway api is not enabled", ch.Namespace, ch.Name)
+			}
```

Comment:
this now fully solves the issue, even if `--enable-gateway-api` is not passed to the controller